### PR TITLE
Add prop to disable datepicker

### PR DESCRIPTION
**Pull request by jherdman**
## Description

Does what it says on the tin.

## Changes include

- [ ] breaking change: a change that is not backwards-compatible and/or changes current functionality
- [x] fix: a non-breaking change that solves an issue
- [x] feature: a non-breaking change that adds functionality
- [ ] chore: contains no changes affecting the library, such as documentation or test updates

## Feature checklist

- [ ] Appropriate tests have been added 
- [ ] Documentation has been updated
- [ ] Accessibility has been considered

### CODE REVIEW

File: src/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -107,9 +112,7 @@ const DatePicker: React.FC<DatePickerProps> = forwardRef(
 
     const handleEnterKey = () => {
       if (ref) {
-        // @ts-ignore
```
Comment (jherdman) [inline src/DatePicker/DatePicker.tsx:110]: I couldn't figure out why these were here. Maybe the type fixes helped?

**Comment by haideralsh** [comment]
Thanks @jherdman for making this change. From looking at the storybook, I can see that we disable the input but we don't change the visual appearance of it. I think we need to change that to be aligned with the other disabled inputs we have like [these inputs here](https://master--5f60c6c285eaad0022dce67f.chromatic.com/?path=/story/components-daterange--disabled).

File: src/DatePicker/DatePicker.story.tsx [hunk 2]
```diff
@@ -170,3 +170,19 @@ export const UsingRefToControlFocus = () => {
...
+    onChange={action("date changed")}
+    onFocus={action("date selector focused")}
+    onBlur={action("date selector blurred")}
+    onInputChange={action("input changed")}
+    inputProps={{ labelText: "Expiry Date" }}
+    disabled
+  />
+);
+
+Default.story = {
```
Comment (haideralsh) [inline src/DatePicker/DatePicker.story.tsx]: I think you meant `Disabled.story = {...}` ?

Comment (jherdman) [reply src/DatePicker/DatePicker.story.tsx]: smh...

**Pull request closed without merging**